classes/Combination.php

```diff
@@ -114,6 +114,30 @@ class CombinationCore extends ObjectModel
         ],
     ];
 
+    /**
+     * @param null $id
+     * @param null $id_lang
+     * @param null $id_shop
+     */
+    public function __construct($id = null, $id_lang = null, $id_shop = null)
```

Comment:
```suggestion
    public function __construct(?int $id = null, ?int $id_lang = null, ?int $id_shop = null)
```

Reply:
@zuk3975 I just noticed `ObjectModel::__construct()` is as follows
```
public function __construct($id = null, $id_lang = null, $id_shop = null, $translator = null)
```
Then by overriding this constructor we prevent people from controlling the `$translator` argument passed to the function.

Reply:
:+1: